doc/client_design_doc.md

```diff
@@ -0,0 +1,52 @@
+# SQLFlow Client Design Doc
+
+## Overview
+
+SQLFlow Client implements a minimum client that issues queries to and fetch results/logs from [sqlflowserver](https://github.com/wangkuiyi/sqlflowserver).
+
+In `pysqlflow`, the client will be wrapped in the magic command where user directly writes SQL statements.
+
+## Example
+
+```python
+import sqlflow
+
+client = sqlflow.Client(server_url='localhost:50051')
+
+# Regular SQL
+response = client.query('select * from table')
+for dataframe in response:
+    print(dataframe) # {'y': [4, 5, 6], 'x': [1, 2, 3]}
+
+# ML SQL, prints epoch = ..., loss = ...
+client.query('select ... train ...')
+```
+
+## Implementation
+
+`sqlflow.Client` uses grpc to contact the `sqlflowserver`. The service protocol is defined at [sqlflow.proto](https://github.com/wangkuiyi/sqlflowserver/blob/develop/sqlflow.proto).
+
+`sqlflow.Client.__init__` establishes a grpc stub/channel based on `server_url`.
+
+`sqlflow.Client.query` takes a sql statement
```

Comment:
May we change query to execute or run as this sql is not just for query?

Reply:
good idea.